src/controller.js

```diff
@@ -0,0 +1,100 @@
+// controller.js
+
+import fetchJSON from './util';
+import apiKey from './constants';
+
+export default class Controller {
+  constructor(model, view) {
+    this.model = model;
+    this.view = view;
+  }
+
+  selectCategory(event) {
+    const category = event.target.dataset.name;
+    this.model.setCategory(category);
+    this.requestData({
+      category: this.model.category,
+      numberOfRecords: this.model.numberOfRecords,
+    });
+  }
+
+  toggleTopHeadlines() {
+    this.model.setTop(!this.view.cbx.checked);
+    this.requestData(this.model.getModel());
+  }
+
+  onNumberInputChange(event) {
+    if (!event.keyCode || (event.keyCode && event.keyCode === 13)) {
+      this.model.setNumberOfRecords(event.target.value);
+      this.requestData(this.model.getModel());
+    }
+  }
+
+  toggleSearchButton() {
+    this.view.toggleSearchButton();
+  }
+
+  toggleMenu() {
+    this.view.toggleMenu();
+  }
+
+  onSearchInputChange(event) {
+    const searchedSourcesNames = this.model.sourcesNames
+      .filter(sourceName => sourceName.toLowerCase()
+        .indexOf(event.target.value.toLowerCase()) === 0);
```

Comment:
wrong indent, it should be at the same level with `filter`

Reply:
indexOf inside filter